README.md

```diff
@@ -428,9 +501,65 @@ hook.remove(name, hookMethod)
 Example
 
 ```js
-hook.remove('save', validateRecord)
+collection.remove('save', validateRecord)
 ```
 
+## TypeScript
+
+This library contains type definitions for TypeScript. When you use TypeScript we highly recommend using the `Hook.Singular` constructor for your hooks as this allows you to pass along type information for the options object. For example:
+
+```ts
+
+import * as Hook from 'before-after-hook'
```

Comment:
I think `import Hook from 'before-after-hook'` should work, too, due to `module.exports.Hook = Hook`?

Reply:
No, that requires a default export:
![image](https://user-images.githubusercontent.com/6368074/53628420-4adfd280-3c0b-11e9-978b-34b6b693163d.png)

But I rarely see imports like that in TypeScript, so I don't think we should want that.

Reply:
I've taken a closer look at the definitions and in my "Improved TypeScript usage" commit I made a few changes. Your choices when importing the module:
![image](https://user-images.githubusercontent.com/6368074/53631490-ede81a80-3c12-11e9-84e2-05b9df3aea5b.png)

I prefer
```ts
new Singular<any>()
```
more than the
```ts
new Hook.Singular<any>()
```
syntax. That's definitely an opinion of mine, but I suspect others might favour it too.

Plus, you can still do this to make it more readable, call it a descriptive middleground 😉 
```ts
const myHook: HookSingular = new Singular<any>()
```

Reply:
I think I’d like to keep `export = Hook` for the folks that don’t use Typescript. I’d like to support

```js
const Hook = require('before-after-hook')
```